teuthology/orchestra/run.py

```diff
@@ -221,6 +233,73 @@ def __repr__(self):
             name=self.hostname,
             )
 
+class ErrorScanner(object):
+    """
+    Scan for unit tests errors in teuthology.log 
+    """
+    __flag__ = 0
+    def __init__(self):
+        self.ERROR_PATTERNS = {
```

Comment:
I'm curious why `__flag__` is a class var but `ERROR_PATTERNS` is an instance var. I think it would make more sense if it were the other way around, since:

1. `ERROR_PATTERNS` doesn't appear to ever be modified, and contains `re.compile()` statements. Using `re.compile()` is a good choice, and it is not at all expensive to run a few times, but there's no real reason to do this multiple times in a given process.
2. `__flag__` does actually get modified, and multiple `ErrorScanner` instances will conflict when modifying it.

Reply:
Since each job has it's own log file, `__flag__` should be shared between all processes in a job which is why I declared it as a class var. 
Suppose if we make `__flag__` an instance variable and create an `ErrorScanner` object in `RemoteProcess`, then flag would be bound to a process's scope and reinitialize it to 0 every time. This would defeat the purpose of tracking flag.
To test that flag as a class var is enclosed in each job's scope, I made [these changes](https://github.com/VallariAg/teuthology/commit/fa7f5bd40d60239867ee58c3a50f5026aafd559e) and the [logs](https://pulpito.ceph.com/vallariag-2022-06-25_13:05:19-rbd:basic-main-distro-default-smithi/) proved that changes to flag are persistent between processes and do not affect other jobs.  

`ERROR_PATTERNS` should indeed be a class var, makes sense!